### PR TITLE
【Prim】Add multiply,expand,div vjp rules

### PR DESCRIPTION
**Pull request by cxxly**
### PR types

Others
### PR changes

Others
### Describe

- add multiply,expand,div vjp rules
- enhance GetReduceDims to support bidirectional broadcast , for example x=(1 2) y=(2 1), reduce_axes for x is 0, for y is 1
- fix relative elementwise operators with bidirectional broadcast

### CODE REVIEW

File: paddle/fluid/prim/api/manual/utils/utils.h [hunk 1]
```diff
@@ -36,43 +36,48 @@ paddle::experimental::Tensor empty_like(const paddle::experimental::Tensor& x,
...
-              x_dims,
-              y_dims,
-              x_dims[i + bat],
-              y_dims[i],
-              i));
+
+  // Reduced axes
+  for (std::size_t i = 0; i < src_shape.size(); i++) {
+    if (src_shape[i] == 1 && dst_shape[i] > 1) {
+      result.push_back(i);
```
Comment (JiabinYang) [inline paddle/fluid/prim/api/manual/utils/utils.h]: This may be miss some case if x, y has 1 as extra rank

**Review by JiabinYang — COMMENTED** [review]
one comments

File: paddle/fluid/prim/api/manual/utils/utils.h (hunk 1 above)
Comment (cxxly) [reply paddle/fluid/prim/api/manual/utils/utils.h]: done

**Review by JiabinYang — APPROVED** [review]
LGTM

**Pull request merged**